sklearn/dummy.py

```diff
@@ -117,6 +117,9 @@ def fit(self, X, y, sample_weight=None):
 
         self.sparse_output_ = sp.issparse(y)
 
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
```

Comment:
I think we need to be liberal wrt `X`: `DummyClassifier` will often substitute for a pipeline / grid search, where `X` may be a list of strings, a dataframe, a list of open files, or other mess. Don't demand too much of it. Here you ensure it is 2d and numeric and of a particular format, but why?